app/Main.hs

```diff
@@ -90,6 +95,20 @@ parseOptions =
 
     noFail = switch (long "no-fail" <> help "Don't exit with a failure status code when any rule is violated")
 
+    noFailCutoff =
+      option
+        (maybeReader toNofailSeverity)
+        ( short 't'
+            <> long "failure-theshold"
+            <> help "Exit with failure code only when rules with a severity \
+                    \above THRESHOLD are violated. Accepted values: \
```

Comment:
According to my tests setting THRESHOLD to `info` will cause exit code 1 when rules with a severity above **or equal** to `info` are violated. So I'd suggest to change to docs to:
```
            <> help "Exit with failure code only when rules with a severity \
                    \equal to or above THRESHOLD are violated. Accepted values:\
                    \[error | warning | info | style | ignore | none]"
```

Reply:
See https://github.com/hadolint/hadolint/pull/639

Reply:
Thanks for the bug report and the PR. But could you please detail how you determined this, since there was a typo in the long option until today and I can't reproduce the finding on the current master branch:
```
│ ~/repos/hadolint │ᐅ hadolint --failure-threshold=info ~/tmp/Dockerfile
/home/moritz/tmp/Dockerfile:3 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.
│ ~/repos/hadolint │ᐅ echo $?
0
│ ~/repos/hadolint │ᐅ 
(master) / 3aba485 11 hours ago
```
with Dockerfile
```Dockerfile
FROM scratch
RUN a
RUN b
```
Unfortunately, we currently don't have integration tests set up for this.

Reply:
I used the docker container with version 2.4.0 (and typo) for testing this. The Dockerfile is identical to yours.

```cmd
$ docker run --rm -it -v %cd%/Dockerfile:/tmp/Dockerfile hadolint/hadolint:2.4.0 /bin/hadolint --failure-theshold info /tmp/Dockerfile & echo %errorlevel%
/tmp/Dockerfile:4 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.
1

$ docker run --rm -it -v %cd%/Dockerfile:/tmp/Dockerfile hadolint/hadolint:2.4.0 /bin/hadolint --failure-theshold warning /tmp/Dockerfile & echo %errorlevel%
/tmp/Dockerfile:4 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.
0
```

Checking again this is not always reproducible for me. When first executing hadolint with `--failure-theshold ignore` (exit code 1) and then with `--failure-theshold info` it exits with code 1 at first. When running with `info` again it gives exit code 0. Am I doing something wrong or is this an issue with the Windows command line?

I can not reproduce this strange behaviour in WSL Ubuntu (or Powershell), so this is probably not caused by hadolint.
```bash
$ docker run --rm -it -v $(pwd)/Dockerfile:/tmp/Dockerfile hadolint/hadolint:2.4.0 /bin/hadolint --failure-theshold info /tmp/Dockerfile; ec
ho $?
```

Reply:
After sleeping on it for a night, IMHO the current implementation is not ideal from a usability perspective: When I want hadolint to fail on `info` rules, I would have to consult the severity list and lookup the next lower level (which is `style`). Also the word `failure` in `--failure-threshold` implies to me that hadolint would fail when a rule with the given severity is violated.

I'd suggest du adjust the implementation to fail on the given severity and change the default value of `--failure-threshold` from `info` to `warning`. What do you think @m-ildefons, @lorenzo?

Reply:
I deliberately chose the current behaviour, because this let me reuse the rule severities that we already have and still map all necessary behaviours: from never failing to failing on any rule violation. But your are technically correct that the dictionary meaning of the word *threshold* implies that the change in behaviour occurs *at* and not just above that level.

With https://github.com/hadolint/hadolint/pull/639 merged, now program behaviour is not reflected in the docs anymore.
I'm still reluctant to change this now because of the implication to current users, whose CI pipelines might break. I'd like to defer this decision to @lorenzo .